app/helpers/application_helper.rb

```diff
@@ -298,11 +298,12 @@ def expand_contract_shuffle(list_id, shuffle=true)
   def autocomplete_options(method, options={})
     {
       class: "autocomplete",
-      autocomplete_method: (method.is_a?(Array) ? method.to_json : "/autocomplete/#{method}"),
-      autocomplete_hint_text: ts("Start typing for suggestions!"),
-      autocomplete_no_results_text: ts("(No suggestions found)"),
-      autocomplete_min_chars: 1,
-      autocomplete_searching_text: ts("Searching...")
+      # We don't use data: {} to ensure values will be correctly overridden
```

Comment:
I think we can use `data: {}`, as long as we also use it in all the overriding places, and we switch `merge` later in this method to [`deep_merge`](https://api.rubyonrails.org/classes/Hash.html#method-i-deep_merge) which merges recursively.

Reply:
I will give `deep_merge` a shot! The first two things alone weren't enough.